### ISSUE TITLE
"Session already started" at login

### ISSUE THREAD

**Issue by bergwerkweb** [multidots/cakephp-rest-api#23]
Hey there,

after installing and testing this fantastic Rest-API we have trouble logging in on the "standard"-side of our application (=login form for users).
We're Using the Cake ACL Plugin: https://github.com/cakephp/acl

Is this a known error? I saw that there was a fix in september, but it does not seem to solve our problem?

Thanks in advance!

**Comment by traeH-divaD** [comment]
@bergwerkweb  I got the same error "Session already started"  but I'm using database session are you also using Database session?

**Comment by bergwerkweb** [comment]
@traeH-divaD
No, we're using the Standard PHP-Session at the moment.

**Comment by traeH-divaD** [comment]
I'm trying to figure out may be rest-api session_start() calling it twice.

**Comment by narendravaghela** [comment]
@traeH-divaD Did you find the issue with plugin regarding session_start()?

**Comment by traeH-divaD** [comment]
**@narendravaghela**  Nope i'm still pending on this case. I will check the issue on my free time.

**Comment by LorenzoRogai** [comment]
Getting this error too, it seems to load the Rest Api Middleware even outside the api routes

![image](https://user-images.githubusercontent.com/1665768/35779566-4a56452a-09cf-11e8-8d0f-59c49474ac8e.png)

Could you please check?

**Comment by narendravaghela** [comment]
It calls the middleware but does not attach the error handler if the controller is not extending the ApiController.
Could you please elaborate more on your use case @LorenzoRogai?

**Comment by wilsmex** [comment]
This still seems related to the AppController::initialize being called twice I filed last year.  In the latest version, my initialize method is still being called twice with this plugin and "bootstrap => true".  Any update?

**Comment by wilsmex** [comment]
It looks like your manually creating a new instance of the current controller in the "RestApiMiddleware" to see if it is a subclass of your plugin controller ` $controller = ($className) ? new $className() : null;`  By creating that new instance you're forcing an additional main AppController call, which is causing many of these issues it seems.

Not sure on a solution, but perhaps allow the plugin to conditionally load the middleware in their own routes ala: https://stackoverflow.com/questions/47714940/cakephp-3-5-6-disable-csrf-middleware-for-controller

Thoughts?

**Comment by narendravaghela** [comment]
@wilsmex I agree with you, will check and update once I get some time from my day job.
